canopy_bioemi_mod.F90

```diff
@@ -0,0 +1,310 @@
+
+!-----------------------------------------------------------------------
+
+! Description:
+!     computes parameterized canopy biogenic emissions
+
+! Preconditions:
+!     in-canopy FCLAI, model LAI, surface/skin temperature.
+
+! Subroutines and Functions Called:
+
+! Revision History:
+!     Prototype 06/22 by PCC, based on Clifton et al. (2022) algorithms
+! Citation:
+! Clifton, O. E. et al. (2022). Large eddy simulation for investigating
+! coupled forest canopy and turbulence influences on atmospheric chemistry.
+! Journal of Advances in Modeling Earth Systems, 14, e2022MS003078.
+! https://doi.org/10.1029/2022MS003078
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+!     Jan 2023 P.C. Campbell: Initial canopy isoprene only version
+!     Feb 2023 P.C. Campbell: Modified for multiple biogenic species
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+        use canopy_const_mod, ONLY: rk,rgasuniv   !constants for canopy models
+        use canopy_utils_mod,  ONLY: interp_linear1_internal
+        use canopy_phot_mod
+        use canopy_bioparm_mod
+
+! Arguments:
+!     IN/OUT
+        REAL(RK),    INTENT( IN )       :: ZK(:)           ! Input model heights (m)
+        REAL(RK),    INTENT( IN )       :: FCLAI(:)        ! Input Fractional (z) shapes of the
+        ! plant surface distribution (nondimensional), i.e., a Fractional Culmulative LAI
+        REAL(RK),    INTENT( IN )       :: FCH             ! Model input canopy height (m)
+        REAL(RK),    INTENT( IN )       :: LAI             ! Model input total Leaf Area Index
+        REAL(RK),    INTENT( IN )       :: CLU             ! Model input Clumping Index
+        REAL(RK),    INTENT( IN )       :: COSZEN          ! Model input Cosine Solar Zenith Angle
+        REAL(RK),    INTENT( IN )       :: SFCRAD          ! Model input Instantaneous surface downward shortwave flux (W/m2)
+        REAL(RK),    INTENT( IN )       :: TEMP2           ! Model input 2-m Temperature (K)
+        INTEGER,     INTENT( IN )       :: LU_OPT          ! integer for LU type from model mapped to Massman et al. (default = 0/VIIRS)
+        INTEGER,     INTENT( IN )       :: VTYPE           ! Grid cell dominant vegetation type
+        REAL(RK),    INTENT( IN )       :: MODRES          ! Canopy model input vertical resolution (m)
+        REAL(RK),    INTENT( IN )       :: CCE             ! MEGAN Canopy environment coefficient.
+        INTEGER,     INTENT( IN )       :: EMI_IND         ! Input biogenic emissions index
+        REAL(RK),    INTENT( OUT )      :: EMI_OUT(:)      ! Output canopy layer volume emissions (kg m-3 s-1)
+
+! Local Variables
+        REAL(RK) :: RJCF(SIZE(ZK))                 ! Photolysis correction factor for sun/shade fraction of leaf layer
```

Comment:
I would call this something that emphasizes sunlit leaf fraction, perhaps `fsun`, since that is how it is being used.

Reply:
@zmoon Totally agree. Making the change.